src/popup/target-tab-finder.ts

```diff
@@ -40,9 +40,15 @@ export class TargetTabFinder {
                     },
                 );
             } else {
-                this.browserAdapter.getTab(tabIdInUrl, (tab: Tab) => {
-                    resolve(tab);
-                });
+                this.browserAdapter.getTab(
+                    tabIdInUrl,
+                    (tab: Tab) => {
+                        resolve(tab);
+                    },
+                    () => {
+                        throw new Error(`Tab with Id ${tabIdInUrl} not found`);
```

Comment:
This should instead be `reject`ing with the message in question.

Reply:
updated!